backend/services/userProfile.service.js

```diff
@@ -0,0 +1,97 @@
+/* eslint-disable func-names */
+const mongoose = require('mongoose');
+const UserProfile = require('../models/userProfile.model');
+const modificationLogService = require('./modificationLog.service');
+const DatabaseError = require('../errors/database.error');
+const DomainError = require('../errors/domain.error');
+
+const UserProfileService = {};
+
+UserProfileService.createUser = async function (userProfileData, authorEmail) {
+    
+    let newRecord = null;
+    
+    if(Object.prototype.hasOwnProperty.call(userProfileData, "signupEmail") &&
+        userProfileData.signupEmail) {
+        userProfileData.signupEmail = userProfileData.signupEmail.toLowerCase();
+    }
+
+    await mongoose.connection.transaction(async () => {
+    
+        try 
+        {
+            newRecord = await UserProfile.create(userProfileData);
+
+            await modificationLogService.saveLog(authorEmail, 
```

Comment:
please fix the function call format to 
```
await modificationLogService
  .saveLog(authorEmail, newRecord._id, "UserProfile", newRecord );
```

Reply:
I used prettier, but this line was kept as is, as it was less than 100 chars. long....I manually made it this format though, as requested.